src/jit/lsraxarch.cpp

```diff
@@ -1978,13 +1978,15 @@ void LinearScan::TreeNodeInfoInitIntrinsic(GenTree* tree)
             }
             break;
 
-#ifdef _TARGET_X86_
+#if defined(_TARGET_AMD64_) || defined(_TARGET_X86_)
         case CORINFO_INTRINSIC_Cos:
         case CORINFO_INTRINSIC_Sin:
```

Comment:
Found the issue. I forgot to add cases for `Ceiling` and `Floor` here.

`Cos` and `Sin` should still be under `_TARGET_X86_` and calling `NYI_X86`, as before.

While `Round`, `Ceiling`, and `Floor` should exist on all targets (AMD64 and X86) and should only call NYI_X86 on the legacy backend.